service/service.hpp

```diff
@@ -299,7 +336,7 @@ namespace raven
         CHECK(service_.clean_socket());
     }
 
-    static void test_client_server_communication(json::json request, json::json expected_answer) noexcept
+    static void test_client_server_communication(json::json &&request, json::json &&expected_answer) noexcept
```

Comment:
Same as above about rvalue refs

Reply:
for this function, the reason is that is the last function that's use both parameters, that's why i use move semantics here to take the ownership, since we didn't use it anymore after this function.